### PR TITLE
ci: add gpu benchmark endtoend test

### PR DESCRIPTION
**Pull request by liuhatry**

### CODE REVIEW

File: .buildkite/scripts/benchmark_master.sh [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    python -c "import sys; sys.exit(1) if float($img_per_sec) < float($speed) else print('imag_per_sec is bigger than $speed.')"
+}
+
+export HOME=/workdir
+cd /workdir && pip install . && git clone https://github.com/BaguaSys/examples.git
+curl https://sh.rustup.rs -sSf | sh -s -- --default-toolchain stable -y
+source $HOME/.cargo/env
+pip install git+https://github.com/BaguaSys/bagua-core@master
+
+algorithms=(gradient_allreduce bytegrad decentralized)
```
Comment (NOBLES5E) [inline .buildkite/scripts/benchmark_master.sh:28]: we also have qadam and low precision decentralized?

**Review by NOBLES5E — CHANGES_REQUESTED** [review]
see comments

File: .buildkite/scripts/benchmark_master.sh (hunk 1 above)
Comment (liuhatry) [reply .buildkite/scripts/benchmark_master.sh]: done

**Comment by liuhatry** [comment]
done

**Pull request merged**